packages/toolbox-core/src/toolbox_core/tool.py

```diff
@@ -89,7 +89,13 @@ def __init__(
         self.__params = params
         self.__pydantic_model = params_to_pydantic_model(name, self.__params)
 
-        inspect_type_params = [param.to_param() for param in self.__params]
+        # Separate parameters into required (no default) and optional (with
+        # default) to prevent the "non-default argument follows default
+        # argument" error when creating the function signature.
+        required_params = [p for p in self.__params if p.required]
+        optional_params = [p for p in self.__params if not p.required]
+        ordered_params = required_params + optional_params
+        inspect_type_params = [param.to_param() for param in ordered_params]
```

Comment:
nit: you can use [generator expressions](https://peps.python.org/pep-0289/) and [itertools.chain ](https://docs.python.org/3/library/itertools.html#itertools.chain)to avoid additional memory allocations 
```suggestion
        required_params = (p for p in self.__params if p.required)
        optional_params = (p for p in self.__params if not p.required)
        ordered_params = itertools.chain(required_params, optional_params)
        inspect_type_params = [param.to_param() for param in ordered_params]
```

Reply:
Thanks for the suggestion!